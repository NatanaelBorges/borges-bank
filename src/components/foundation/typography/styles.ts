import styled, { css, DefaultTheme } from 'styled-components';
import { TypographyProps } from '.';

const textSize = {
  h1: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.huge};
  `,
  h2: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.large};
  `,
  h3: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.big};
  `,
  h4: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.medium};
  `,
  h5: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.small};
  `,
  h6: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.xsmall};
  `,
};

const textSizeParagraph = {
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.big};
  `,
};

const textCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const textItalic = (italic: boolean) => css`
  font-style: ${italic ? 'italic' : 'none'};
`;

const textUnderline = (underline: boolean) => css`
  text-decoration: ${underline ? 'underline' : 'none'};
`;

const textBold = (bold: boolean) => css`
  font-weight: ${bold ? 'bold' : 'normal'};
`;

const textColor = {
  primary: (theme: DefaultTheme) => css`
    color: ${theme.colors.primary};
  `,
  secondary: (theme: DefaultTheme) => css`
    color: ${theme.colors.secondary};
  `,
  white: (theme: DefaultTheme) => css`
    color: ${theme.colors.white};
  `,
  black: (theme: DefaultTheme) => css`
    color: ${theme.colors.black};
  `,
};

const textDisplay = {
  display1: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.huge};
  `,
  display2: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.large};
  `,
  display3: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.big};
  `,
  display4: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.medium};
  `,
  display5: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.small};
  `,
  display6: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.xsmall};
  `,
};

export const Text = styled.h1<TypographyProps>`
  ${({
    theme,
    align,
    italic,
    uppercase,
    as,
    display,
    size,
    underline,
    bold,
    color,
  }) => css`
    text-align: ${align};
    ${textCase(uppercase)};
    ${textColor[color](theme)};
    ${textUnderline(underline)};
    ${textItalic(italic)};
    ${textBold(bold)};
    ${as === 'p' ? textSizeParagraph[size](theme) : textSize[as](theme)};
    ${display !== 'none' ? textDisplay[display](theme) : 'none'};
  `}
`;
