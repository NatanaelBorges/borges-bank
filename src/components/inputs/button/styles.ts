import styled, { css } from 'styled-components';
import { Link } from 'react-scroll';
import { ButtonProps } from '.';

export const Button = styled.button<ButtonProps>`
  ${({ theme, fontBig, big, primary, dark }) => css`
    border-radius: 50px;
    background: ${primary ? theme.colors.primary : theme.colors.dark};
    white-space: nowrap;
    padding: ${big ? '14px  48px' : '12px 30px'};
    color: ${dark ? theme.colors.dark : theme.colors.white};
    font-size: ${fontBig ? '20px' : '16px'};
    outline: none;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: all 0.2s ease-in-out;

    &:hover {
      transition: all 0.2s ease-in-out;
      background: ${primary ? theme.colors.white : theme.colors.primary};
    }
  `}
`;

export const href = styled(Link)``;
