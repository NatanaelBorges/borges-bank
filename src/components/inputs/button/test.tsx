import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import { theme } from '../../../styles/theme';

import { Button } from '.';

describe('<Button />', () => {
  it('should render with default values', () => {
    renderTheme(<Button>text</Button>);
    const button = screen.getByRole('button', { name: 'text' });

    expect(button).toHaveStyle({
      'border-radius': '50px',
      background: theme.colors.primary,
      'white-space': 'nowrap',
      padding: '12px 30px',
      color: theme.colors.white,
      'font-size': '16px',
      outline: 'none',
      border: 'none',
      cursor: 'pointer',
    });
  });

  it('should render with sizes big', () => {
    renderTheme(<Button big={true}>text</Button>);
    const button = screen.getByRole('button', { name: 'text' });

    expect(button).toHaveStyle({
      padding: '14px 48px',
    });
  });

  it('should render with font big', () => {
    renderTheme(<Button fontBig={true}>text</Button>);
    const button = screen.getByRole('button', { name: 'text' });

    expect(button).toHaveStyle({
      'font-size': '20px',
    });
  });

  it('should render correct colors', () => {
    renderTheme(<Button primary={false}>text</Button>);
    const button = screen.getByRole('button', { name: 'text' });

    expect(button).toHaveStyle({
      background: theme.colors.dark,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Button>text</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
