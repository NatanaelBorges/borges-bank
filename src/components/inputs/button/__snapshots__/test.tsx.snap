// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
.c0 {
  border-radius: 50px;
  background: #01BF71;
  white-space: nowrap;
  padding: 12px 30px;
  color: #FFFFFF;
  font-size: 16px;
  outline: none;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background: #FFFFFF;
}

<button
  class="c0"
>
  text
</button>
`;
